### PR TITLE
include log class path as liquibase property

### LINKED ISSUE #7005: LiquibaseConfiguration: hard coded master.xml
The problem comes from generator-jhipster as this file is really similar

See https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/springboot/dbmigration/liquibase/main/LiquibaseConfiguration.java.mustache#L64

It should not be hard coded, and instead, it should use the property `spring.liquibase.change-log`

It will help the user to override the property instead of modifying the java class

To do:
- use the property from `spring.liquibase.change-log` instead of hard coded `master.xml`
- add in application.properties: `spring.liquibase.change-log=classpath:config/liquibase/master.xml`

### PR DESCRIPTION
**Pull request by adem-hassine**
Change resolves #7005 by modifying the mustache file & it's liquibase factory to include required env variable

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactory.java [hunk 1]
```diff
@@ -29,6 +29,9 @@ public JHipsterModule buildModule(JHipsterModuleProperties properties) {
         .addDependency(groupId("org.liquibase"), artifactId("liquibase-core"), versionSlug(LIQUIBASE))
         .addDependency(h2Dependency())
         .and()
+      .springMainProperties()
+      .set(propertyKey("spring.liquibase.change-log"), propertyValue("classpath:config/liquibase/master.xml"))
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactory.java:33]: can you indent this line and the next one with two spaces so it's easier to see that it is for springMainProperties

**Comment by DamnClin** [comment]
Thanks a lot for your first contribution. Sorry for the review delay...

File: src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactory.java (hunk 1 above)
Comment (adem-hassine) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactory.java]: Sure, I just submitted a new commit for that.

**Comment by adem-hassine** [comment]
No issues! I'm delighted to make a contribution to this project.

**Comment by DamnClin** [comment]
Looks like you have a coverrage drop : 

```
Warning:  Rule violated for class tech.jhipster.mariadbapp.security.infrastructure.primary.CorsFilterConfiguration: lines covered ratio is 0.00, but expected minimum is 1.00
Warning:  Rule violated for class tech.jhipster.mariadbapp.security.infrastructure.primary.CorsFilterConfiguration: branches covered ratio is 0.00, but expected minimum is 1.00
Warning:  Rule violated for class tech.jhipster.mariadbapp.security.infrastructure.primary.CorsProperties: lines covered ratio is 0.00, but expected minimum is 1.00
Warning:  Rule violated for class tech.jhipster.mariadbapp.technical.infrastructure.secondary.liquibase.AsyncSpringLiquibase: branches covered ratio is 0.83, but expected minimum is 1.00
Warning:  Rule violated for class tech.jhipster.mariadbapp.technical.infrastructure.secondary.liquibase.LiquibaseConfiguration: lines covered ratio is 0.13, but expected minimum is 1.00
Warning:  Rule violated for class tech.jhipster.mariadbapp.technical.infrastructure.secondary.liquibase.LiquibaseConfiguration: branches covered ratio is 0.00, but expected minimum is 1.00
Warning:  Rule violated for class tech.jhipster.mariadbapp.technical.infrastructure.secondary.mariadb.DatabaseConfiguration: lines covered ratio is 0.00, but expected minimum is 1.00
```

I haven't looked at it in details. Does that seem related to your update or you are just unlucky?

**Comment by adem-hassine** [comment]
I don't believe it's related to my update since I didn't make any modifications to the files mentioned in the warning trace. The coverage is significantly lower than expected, and honestly, I'm not familiar with the history of these changes, but I don't think it's caused by my recent update.

File: src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactory.java [hunk 2]
```diff
@@ -29,6 +29,9 @@ public JHipsterModule buildModule(JHipsterModuleProperties properties) {
         .addDependency(groupId("org.liquibase"), artifactId("liquibase-core"), versionSlug(LIQUIBASE))
         .addDependency(h2Dependency())
         .and()
+      .springMainProperties()
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactory.java:32]: This should be tested in the factory unit test

**Comment by DamnClin** [comment]
> I don't believe it's related to my update since I didn't make any modifications to the files mentioned in the warning trace. The coverage is significantly lower than expected, and honestly, I'm not familiar with the history of these changes, but I don't think it's caused by my recent update.

In fact this is related to your update :). The test application can't start since the new key is unknown in test context, resulting in some test not being able to run. You need to add a springTestProperties or a default value for the key

File: src/test/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactoryTest.java [hunk 3]
```diff
@@ -27,21 +27,21 @@ void shouldBuildModule() {
       .hasFile("pom.xml")
       .containing(
         """
-            <dependency>
-              <groupId>org.liquibase</groupId>
-              <artifactId>liquibase-core</artifactId>
-              <version>${liquibase.version}</version>
-            </dependency>
-        """
+              <dependency>
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactoryTest.java:30]: Indentation broke here :)

File: src/test/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactoryTest.java [hunk 4]
```diff
@@ -27,21 +27,21 @@ void shouldBuildModule() {
...
       )
       .containing(
         """
-            <dependency>
-              <groupId>com.h2database</groupId>
-              <artifactId>h2</artifactId>
-              <scope>test</scope>
-            </dependency>
-        """
+              <dependency>
```
Comment (DamnClin) [inline src/test/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseModuleFactoryTest.java:39]: Indentation broke here :)

**Comment by adem-hassine** [comment]
> > I don't believe it's related to my update since I didn't make any modifications to the files mentioned in the warning trace. The coverage is significantly lower than expected, and honestly, I'm not familiar with the history of these changes, but I don't think it's caused by my recent update.
> 
> In fact this is related to your update :). The test application can't start since the new key is unknown in test context, resulting in some test not being able to run. You need to add a springTestProperties or a default value for the key

Sorry for the inconvenience, and hopefully everything is fixed now. Thank you for your assistance.

**Comment by DamnClin** [comment]
> Sorry for the inconvenience, and hopefully everything is fixed now. Thank you for your assistance.

No problem :). thanks for your time

**Comment by DamnClin** [comment]
Congratz for your first PR :)

**Pull request merged**